File: ts/src/domain/CarName/CarName.test.ts
import { CarName } from './CarName';

describe('CarName', () => {
    describe('생성', () => {
        it('유효한 이름으로 생성할 수 있다', () => {
            const carName = new CarName('kim');
            expect(carName.getValue()).toBe('kim');
        });

        it('5자 이름으로 생성할 수 있다', () => {
            const carName = new CarName('abcde');
            expect(carName.getValue()).toBe('abcde');
        });
    });

    describe('유효성 검증', () => {
        it('이름이 5자를 초과하면 예외를 던진다', () => {
            expect(() => new CarName('abcdef')).toThrow('자동차 이름은 5자를 초과할 수 없습니다.');
        });

        it('이름이 빈 문자열이면 예외를 던진다', () => {
            expect(() => new CarName('')).toThrow('자동차 이름은 빈 문자열일 수 없습니다.');
        });

        it('이름이 공백만 있으면 예외를 던진다', () => {
            expect(() => new CarName('   ')).toThrow('자동차 이름은 빈 문자열일 수 없습니다.');
        });
    });

    describe('getValue', () => {
        it('설정한 이름 값을 반환한다', () => {
            const carName = new CarName('park');
            expect(carName.getValue()).toBe('park');
        });
    });
});

File: java/src/test/java/racing/CarNameTest.java
package racing;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.assertThatThrownBy;

public class CarNameTest {

    @DisplayName("자동차 이름은 비어있을 수 없다")
    @Test
    void emptyOrNull() {
        assertThatThrownBy(() -> CarName.of(null))
                .isInstanceOf(IllegalArgumentException.class);

        assertThatThrownBy(() -> CarName.of(""))
                .isInstanceOf(IllegalArgumentException.class);
    }

    @DisplayName("자동차 이름은 5자를 초과할 수 없다")
    @Test
    void nameOverMaxLength() {
        assertThatThrownBy(() -> CarName.of("abcdef"))
                .isInstanceOf(IllegalArgumentException.class);
    }

    @DisplayName("자동차 이름은 5자까지 허용된다")
    @Test
    void nameAtMaxLength() {
        // given
        String name = "abcde";

        // when
        CarName carName = CarName.of(name);

        // then
        assertThat(carName).isNotNull();
    }

    @DisplayName("같은 이름의 CarName은 동등하다")
    @Test
    void equality() {
        CarName name1 = CarName.of("kim");
        CarName name2 = CarName.of("kim");

        assertThat(name1).isEqualTo(name2);
        assertThat(name1).hasSameHashCodeAs(name2);
    }
}